File: solutions/javascript/class-design/loggerRateLimiter.js
const COOLDOWN_PERIOD_LENGTH = 10;

class Logger {
    constructor() {
        this.lastPrintTime = {};
    }

    shouldPrintMessage(timestamp, message) {
        const messagePreviouslyPrinted =
            this.lastPrintTime.hasOwnProperty(message);

        if (!messagePreviouslyPrinted) {
            this.lastPrintTime[message] = timestamp;
            return true;
        }

        const timeSinceLastPrint = timestamp - this.lastPrintTime[message];
        if (timeSinceLastPrint < COOLDOWN_PERIOD_LENGTH) return false;

        this.lastPrintTime[message] = timestamp;
        return true;
    }
}

File: solutions/python/class-design/logger_rate_limiter.py
COOLDOWN_PERIOD_LENGTH = 10

class Logger:
    def __init__(self):
        self.last_print_time = {}

    def shouldPrintMessage(self, timestamp, message):
        

        if message not in self.last_print_time:
            self.last_print_time[message] = timestamp
            return True

        time_since_last_print = timestamp - self.last_print_time[message]
        if time_since_last_print < COOLDOWN_PERIOD_LENGTH:
            return False

        self.last_print_time[message] = timestamp
        return True